app/code/Magento/Inventory/Model/SourceRepository.php

```diff
@@ -0,0 +1,120 @@
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceSearchResultsInterface;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\Inventory\Model\SourceSearchResultsFactory;
+use Magento\Inventory\Model\Resource\Source as ResourceSource;
+use Magento\Inventory\Model\Resource\Source\CollectionFactory;
+use Magento\Inventory\Model\SourceFactory;
+
+/**
+ * Class SourceRepository
+ * @package Magento\Inventory\Model
+ */
+class SourceRepository implements SourceRepositoryInterface
+{
+    /**
+     * @var ResourceSource
+     */
+    private $resource;
+
+    /**
+     * @var SourceFactory
+     */
+    private $sourceFactory;
+
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * @var SourceSearchResultsFactory
+     */
+    private $sourceSearchResultsFactory;
+
+    /**
+     * SourceRepository constructor.
+     * @param ResourceSource $resource
+     * @param SourceFactory $sourceFactory
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param CollectionFactory $collectionFactory
+     * @param SourceSearchResultsFactory $sourceSearchResultsFactory
+     */
+    public function __construct(
+        ResourceSource $resource,
+        SourceFactory $sourceFactory,
+        CollectionProcessorInterface $collectionProcessor,
+        CollectionFactory $collectionFactory,
+        SourceSearchResultsFactory $sourceSearchResultsFactory
+    ) {
+        $this->resource = $resource;
+        $this->sourceFactory = $sourceFactory;
+        $this->collectionProcessor = $collectionProcessor;
+        $this->collectionFactory = $collectionFactory;
+        $this->sourceSearchResultsFactory = $sourceSearchResultsFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(SourceInterface $source)
+    {
+        try {
+            $this->resource->save($source);
+        } catch (\Exception $exception) {
+            throw new CouldNotSaveException(__($exception->getMessage()));
+        }
+        return $source;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get($sourceId)
+    {
+        /** @var SourceInterface|AbstractModel $model */
+        $model = $this->sourceFactory->create();
+        $this->resource->load($model, SourceInterface::SOURCE_ID, $sourceId);
+
+        if (!$model->getSourceId()) {
+            throw NoSuchEntityException::singleField(SourceInterface::SOURCE_ID, $sourceId);
+        }
+
+        return $model;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getList(SearchCriteriaInterface $searchCriteria = null)
+    {
+        /** @var \Magento\Inventory\Model\Resource\Source\Collection $collection */
+        $collection = $this->collectionFactory->create();
+        $this->collectionProcessor->process($searchCriteria, $collection);
```

Comment:
have nullable $searchCriteria, you will pass null to Collection Processor.
Currently interface of Collection Processor looks like:
```php
/**
 * @api
 */
interface CollectionProcessorInterface
{
    /**
     * Apply Search Criteria to Collection
     *
     * @param SearchCriteriaInterface $searchCriteria
     * @param AbstractDb $collection
     * @throws \InvalidArgumentException
     * @return void
     */
    public function process(SearchCriteriaInterface $searchCriteria, AbstractDb $collection);
}

```
So, interface doesn't support passing null value.